src/Materials/Textures/Packer/packer.ts

```diff
@@ -0,0 +1,792 @@
+import { AbstractMesh } from "../../../Meshes/abstractMesh";
+import { VertexBuffer } from "../../../Meshes/buffer";
+import { Scene } from "../../../scene";
+import { Material } from "../../material";
+import { Texture } from "../texture";
+import { DynamicTexture } from "../dynamicTexture";
+import { Nullable } from "../../../types";
+import { Vector2 } from "../../../Maths/math.vector";
+import { Color3, Color4 } from "../../../Maths/math.color";
+import { TexturePackerFrame } from "./frame";
+
+/**
+* Defines the basic options interface of a TexturePacker
+*/
+export interface ITexturePackerOptions{
+
+    /**
+    * Custom targets for the channels of a texture packer.  Default is all the channels of the Standard Material
+    */
+    map?: string[];
+
+    /**
+    * the UV input targets, as a single value for all meshes. Defaults to VertexBuffer.UVKind
+    */
+    uvsIn?: string;
+
+    /**
+    * the UV output targets, as a single value for all meshes.  Defaults to VertexBuffer.UVKind
+    */
+    uvsOut?: string;
+
+    /**
+    * number representing the layout style. Defaults to LAYOUT_STRIP
+    */
+    layout?: number;
+
+    /**
+    * number of columns if using custom column count layout(2).  This defaults to 4.
+    */
+    colnum?: number;
+
+    /**
+	 * flag to update the input meshes to the new packed texture after compilation. Defaults to true.
+	 */
+    updateInputMeshes?: boolean;
+
+    /**
+	* boolean flag to dispose all the source textures.  Defaults to true.
+	*/
+    disposeSources?: boolean;
+
+    /**
+    * Fills the blank cells in a set to the customFillColor.  Defaults to true.
+    */
+    fillBlanks?: boolean;
+
+    /**
+    * string value representing the context fill style color.  Defaults to 'black'.
+    */
+    customFillColor?: string;
+
+    /**
+    * Width and Height Value of each Frame in the TexturePacker Sets
+    */
+    frameSize?: number;
+
+    /**
+    * Ratio of the value to add padding wise to each cell.  Defaults to 0.0115
+    */
+    paddingRatio?: number;
+
+    /**
+    * Number that declares the fill method for the padding gutter.
+    */
+    paddingMode?: number;
+
+    /**
+    * If in SUBUV_COLOR padding mode what color to use.
+    */
+    paddingColor?: Color3 | Color4;
+
+}
+
+/**
+* Defines the basic interface of a TexturePacker JSON File
+*/
+export interface ITexturePackerJSON{
+
+    /**
+    * The frame ID
+    */
+    name: string;
+
+    /**
+    * The base64 channel data
+    */
+    sets: any;
+
+    /**
+    * The options of the Packer
+    */
+    options: ITexturePackerOptions;
+
+    /**
+    * The frame data of the Packer
+    */
+    frames: Array<number>;
+
+}
+
+/**
+* This is a support class that generates a series of packed texture sets.
+* @see https://doc.babylonjs.com/babylon101/materials
+*/
+export class TexturePacker{
+
+    /** Packer Layout Constant 0 */
+    public static readonly LAYOUT_STRIP = Constants.LAYOUT_STRIP;

```

Comment:
Actually I give that a thought and we do not need to expose the constants in the constant file as they are only used by the packer. So we can keep the declartion here (with the direct value instead of a link to Constants)

Sorry buddy about that last one